pyanaconda/core/users.py

```diff
@@ -43,13 +44,24 @@ def crypt_password(password):
     :returns: crypted representation of the original password
     :rtype: str
     """
-    cryptpw = crypt.crypt(password, crypt.METHOD_SHA512)
+    # yescrypt is not supported by Python's crypt module,
+    # so we need to generate the setting ourselves
+    b64 = "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+    setting = "$y$j9T$" + "".join(sr().choice(b64) for sc in range(24))
 
+    # and compute the password hash using our yescrypt setting
+    cryptpw = crypt.crypt(password, setting)
+
+    # Fallback to sha512crypt, if yescrypt is not supported
     if cryptpw is None:
```

Comment:
`None` as a return value is not documented in https://docs.python.org/3/library/crypt.html#crypt.crypt, so I was playing with it and it looks like `crypt.crypt` raises the OSError exception now (https://github.com/python/cpython/commit/0d3fe8ae4961bf551e7d5e42559e2ede1a08fd7c).

```
>>> crypt.crypt("password", "")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.9/crypt.py", line 82, in crypt
    return _crypt.crypt(word, salt)
OSError: [Errno 22] Invalid argument

```

Reply:
Indeed!  Thanks for the catch!

Reply:
Addressed in 8781132.